tests/functional/oauth_webchannel_keys.js

```diff
@@ -0,0 +1,510 @@
+        })
+
+        .switchToWindow('newwindow')
+        .execute(FunctionalHelpers.listenForWebChannelMessage)
+
+        .then(testIsBrowserNotifiedOfLogin(self, { shouldCloseTab: false }))
+
+        .findById('fxa-sign-up-complete-header')
+        .end()
+
+        .closeCurrentWindow()
+        // switch to the original window
+        .switchToWindow('');
+    },
+
+    'signup, verify same browser, replace original tab': function () {
+      var self = this;
+
+      return openFxaFromRp(self, 'signup')
+
+        .then(function () {
+          return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD, OLD_ENOUGH_YEAR);
+        })
+
+        .findByCssSelector('#fxa-confirm-header')
+        .end()
+
+        .then(function () {
+          return FunctionalHelpers.getVerificationLink(email, 0);
+        })
+        .then(function (verificationLink) {
+          return self.get('remote').get(require.toUrl(verificationLink))
+            .execute(FunctionalHelpers.listenForWebChannelMessage);
+        })
+
+        .then(testIsBrowserNotifiedOfLogin(self, { shouldCloseTab: false }))
+
+        .findById('fxa-sign-up-complete-header')
+        .end();
+    },
+
+    'signup, verify different browser - from original tab\'s P.O.V.': function () {
+      var self = this;
+
+      return openFxaFromRp(self, 'signup')
+        .execute(FunctionalHelpers.listenForWebChannelMessage)
+
+        .then(function () {
+          return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD, OLD_ENOUGH_YEAR);
+        })
+
+        .findByCssSelector('#fxa-confirm-header')
+        .end()
+
+        .then(function () {
+          return FunctionalHelpers.openVerificationLinkDifferentBrowser(client, email);
+        })
+
+        .then(testIsBrowserNotifiedOfLogin(self, { shouldCloseTab: false }))
+
+        .findById('fxa-sign-up-complete-header')
+        .end();
+    },
+
+    'signup, verify different browser - from new browser\'s P.O.V.': function () {
+      var self = this;
+
+      return openFxaFromRp(self, 'signup')
+        .execute(FunctionalHelpers.listenForWebChannelMessage)
+
+        .then(function () {
+          return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD, OLD_ENOUGH_YEAR);
+        })
+
+        .findByCssSelector('#fxa-confirm-header')
+        .end()
+
+        .then(function () {
+          // clear browser state to simulate opening the link
+          // in the same browser
+          return FunctionalHelpers.clearBrowserState(self);
+        })
+
+        .then(function () {
+          return FunctionalHelpers.getVerificationLink(email, 0);
+        })
+        .then(function (verificationLink) {
+          return self.get('remote').get(require.toUrl(verificationLink));
+        })
+
+        // new browser dead ends at the 'account verified' screen.
+        // It should not try to generate keys and complete the flow.
+        .findByCssSelector('#fxa-sign-up-complete-header')
+        .end();
+    },
```

Comment:
Yeah, I guess these "from new browser P.O.V." tests aren't really testing anything.  I'm going to remove them here, they're equally well covered by their equivalents in the non-keys webchannel case.